File: src/main/java/org/cloudfoundry/credhub/exceptions/AccessTokenExpiredException.java
package org.cloudfoundry.credhub.exceptions;

import org.springframework.security.oauth2.common.exceptions.ClientAuthenticationException;

public class AccessTokenExpiredException extends ClientAuthenticationException {

  public AccessTokenExpiredException(String msg, Throwable t) {
    super(msg, t);
  }

  @Override
  public String getOAuth2ErrorCode() {
    return "access_token_expired";
  }

  @Override
  public int getHttpErrorCode() {
    return 401;
  }
}

File: components/errors/src/main/kotlin/org/cloudfoundry/credhub/exceptions/AccessTokenExpiredException.kt
package org.cloudfoundry.credhub.exceptions

import org.springframework.security.oauth2.common.exceptions.ClientAuthenticationException

class AccessTokenExpiredException(
    msg: String,
    t: Throwable,
) : ClientAuthenticationException(msg, t) {
    override fun getOAuth2ErrorCode(): String = "access_token_expired"

    override fun getHttpErrorCode(): Int = 401
}
